### PR TITLE
add flag to disable prepared statements (close #1392)

### LINKED ISSUE #1392: allow disabling prepared statements globally
It helps us to debug issues with prepared statements.

### PR DESCRIPTION
**Pull request by nizar-m**
### Description

What component does this PR affect? 

- [x] Server
- [ ] Console
- [ ] CLI
- [ ] Docs
- [ ] Community Content
- [ ] Build System

Requires changes from other components? If yes, please mark the components:

- [ ] Server
- [ ] Console
- [ ] CLI
- [x] Docs
- [ ] Community Content
- [ ] Build System

### Related Issue

#1392 

### Solution and Design

- --use-prepared-statements (OR EnvVar HASURA_GRAPHQL_USE_PREPARED_STATEMENTS) to control whether prepared statements can be used or not
- Set this as the boolean configuration allowPrepare for the Postgres Connection Pool 

### Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Docs update
- [ ] Community content

### Checklist:

- [x] I have read the **[contributing guide](https://github.com/hasura/graphql-engine/blob/master/CONTRIBUTING.md)** and my code conforms to the guidelines.
- [x] This change requires a change in the documentation. 
- [x] I have updated the documentation accordingly.
- [ ] I have added required tests.

### CODE REVIEW

File: server/src-lib/Hasura/Server/Init.hs [hunk 1]
```diff
@@ -361,6 +363,12 @@ pgTimeoutEnv =
   , "Each connection's idle time before it is closed (default: 180 sec)"
   )
 
+pgAllowPrepareEnv :: (String, String)
+pgAllowPrepareEnv =
+  ( "HASURA_GRAPHQL_ALLOW_SQL_PREPARE"
```
Comment (0x777) [inline server/src-lib/Hasura/Server/Init.hs]: `HASURA_GRAPHQL_USE_PREPARED_STATEMENTS`

File: server/src-lib/Hasura/Server/Init.hs [hunk 2]
```diff
@@ -508,6 +516,12 @@ parseConnParams =
                 metavar "SECONDS" <>
                 help (snd pgTimeoutEnv)
               )
+    allowPrepare = optional $
+      option auto
+              ( long "allow-sql-prepare" <>
```
Comment (0x777) [inline server/src-lib/Hasura/Server/Init.hs]: `--use-prepared_statements`

**Comment by 0x777** [comment]
@nizar-m You'll also need to add these options to docs.

**Comment by shahidhk** [comment]
Need docs.

**Pull request merged**